nvim/lua/plugins/lsp.lua

```diff
@@ -52,123 +42,145 @@ return {
             })(entry, item)
           end
         },
-        -- snippet = {
-        --   expand = function(args)
-        --     require('luasnip').lsp_expand(args.body)
-        --   end
-        -- },
+        snippet = {
+          expand = function(args)
+            luasnip.lsp_expand(args.body)
+          end
+        },
+        window = {
+          documentation = cmp.config.window.bordered {
+            winhighlight = 'Normal:NormalFloat,FloatBorder:FloatBorder,CursorLine:Visual,Search:None',
+          },
+        },
         mapping = cmp.mapping.preset.insert({
-          ['<Tab>'] = cmp.mapping.select_next_item(cmp_select),
-          ['<S-Tab>'] = cmp.mapping.select_prev_item(cmp_select),
+          ['<Tab>'] = cmp.mapping.select_next_item(),
+          ['<S-Tab>'] = cmp.mapping.select_prev_item(),
           ['<C-space>'] = cmp.mapping.complete(),
           ['<Esc>'] = cmp.mapping.abort(),
           ['<CR>'] = cmp.mapping.confirm({
             behavior = cmp.ConfirmBehavior.Replace,
             select = true
-          })
+          }),
+
+          -- <c-l> will move you to the right of each of the expansion locations.
+          -- <c-h> is similar, except moving you backwards.
+          ['<C-l>'] = cmp.mapping(function()
+            if luasnip.expand_or_locally_jumpable() then
+              luasnip.expand_or_jump()
+            end
+          end, { 'i', 's' }),
+          ['<C-h>'] = cmp.mapping(function()
+            if luasnip.locally_jumpable(-1) then
+              luasnip.jump(-1)
+            end
+          end, { 'i', 's' }),
         })
       })
     end
   },
 
   {
-    'VonHeikemen/lsp-zero.nvim',
-    branch = 'v3.x',
-    event = { 'BufReadPre', 'BufNewFile' },
+    'neovim/nvim-lspconfig',
     dependencies = {
-      { 'neovim/nvim-lspconfig' },
-      { 'williamboman/mason.nvim' },
+      { 'williamboman/mason.nvim', config = true },
       { 'williamboman/mason-lspconfig.nvim' },
       { 'nvim-cmp' },
+      { 'hrsh7th/cmp-nvim-lsp' }
     },
-    init = function()
-      vim.diagnostic.config({
-        virtual_text = true
-      })
-    end,
-    config = function()
-      local lsp_zero = require('lsp-zero')
-      local mason_lspconfig = require('mason-lspconfig')
+    config = function ()
+      vim.api.nvim_create_autocmd('LspAttach', {
+        group = vim.api.nvim_create_augroup('kickstart-lsp-attach', { clear = true }),
+        callback = function (event)
+          local telescope_builtin = require('telescope.builtin')
+          local map = function(keys, func, desc, mode)
+            mode = mode or 'n'
+            vim.keymap.set(mode, keys, func, { buffer = event.buf, desc = 'LSP: ' .. desc, remap = false })
+          end
+
+          map('gd', telescope_builtin.lsp_definitions, '[G]oto [D]efinition')
+          map('gD', vim.lsp.buf.declaration, '[G]oto [D]eclaration')
 
-      require('mason').setup({})
```

Comment:
It's no longer necessary since we already have it initialized in dependencies section

https://github.com/feryardiant/dotfiles/blob/14d30fadc2d93757c1f67d663ce5131523ecbe93/nvim/lua/plugins/lsp.lua#L85